### ISSUE TITLE
Sony ILCE-7M4

### ISSUE THREAD

**Issue by LucasRueckert** [darktable-org/darktable#12064]
I uploaded a bunch of RAW-Images to raw.pixls.us to improve support for the camera.
Uploaded Fotos:

- Full Frame and APS-C
- in RAW, Lossless Compressed RAW and Compressed RAW
- in 3:2, 4:3, 16:9 and 1:1

**Comment by kmilos** [comment]
Duplicate of https://github.com/darktable-org/darktable/issues/10556

Please search before opening new issues.

Support is already present in 3.8.1, just not for the lossless compression yet.

**Comment by kmilos** [comment]
APS-C might need some extra work, so thanks for those and cropped samples.

**Comment by vancouverbluesea** [comment]
I am not sure this camera is fully supported yet.... I don't have the camera but I did check the samples on https://www.dpreview.com/sample-galleries/9601355361/sony-a7-iv-sample-gallery/4038700934 and many of the files can't open in DT 4.0.0

**Comment by LucasRueckert** [comment]
I can view lossless compressed photos in the Lighttable, but when i click on them to edit, i get an error that the format is not supported. in 4.0.0

**Comment by MStraeten** [comment]
in lighttable the embedded jpg is displayed until you edited the image in darkroom

**Comment by vancouverbluesea** [comment]
> in lighttable the embedded jpg is displayed until you edited the image in darkroom

But it won't open in darkroom.

**Comment by kmilos** [comment]
As mentioned already, lossless is not supported.

**Comment by orestesmas** [comment]
My son bought this camera some weeks ago. I'd like to see support for it in DarkTable. Not being a programmer, is there any other thing that I can do to speed up this process? Upload more images, for instance?

Regards.
Orestes.

**Comment by kmilos** [comment]
The camera is supported, just not in the lossless mode.

**Comment by orestesmas** [comment]
Ok, sorry, I wasn't clear enough in my last comment: My son and I were unaware of this issue and he did a bunch of _uncompressed_ (lossless?) raw photos this August in Switzerland. Returning home, we saw that DarkTable could not open them, hence my comment.

I have been able to convert them to a demosaiced TIFF using the "dcraw_emu" utility of LibRaw 0.21 (beta 1), which recently introduced support for ILCE-7M4. But, honestly, I'm completely ignorant of how to control the demosaicing parameters from this command-line utility.

So, perhaps a more accurate statement would be that DarkTable has already _partial_ support for Sony ILCE-7M4. Thus, I would rephrase my question to: Do ​​you know when this camera will get full raw format support? Can I help somehow?

Thank you.
Orestes

**Comment by kmilos** [comment]
Uncompressed should work. Lossless doesn't. Different things.

> Do ​​you know when this camera will get full raw format support? Can I help somehow?

When someone implements the codec to the rawspeed coding standard.

**Comment by kmilos** [comment]
A better option might be to convert those DNG btw.

**Comment by GustavHaapalahti** [comment]
It does not look like there is any noise profle for sony a7iv in the latest version

**Comment by LebedevRI** [comment]
1b707f345230d6916aa4fc0a33231dcb03153431

**Comment by Ugzuzg** [comment]
The camera has three lossless compression presets: L, M, S. I took a photo in each. Only the L one was opened by darktable 4.4.2.

Should at least one of the issues about lossless compression be kept open, so that it is clear that the support isn't completely there yet?

```
     8,6452 [rawspeed] (20230725_0048.ARW) void rawspeed::ArwDecoder::DecodeLJpeg(const rawspeed::TiffIFD*), line 291: Unsupported photometric interpretation: 6
     8,6452 [libraw_open] extensions whitelist: `cr3'
     8,6466 [temperature] failed to read camera white balance information from `20230725_0048.ARW'!
     8,9812 [temperature] `Sony ILCE-7M4' color matrix not found for image
     8,9812 [temperature] failed to read camera white balance information from `20230725_0048.ARW'!
     8,9817 [temperature] failed to read camera white balance information from `20230725_0048.ARW'!
     8,9932 [colorin] could not find requested profile `standard color matrix'!
```

**Comment by kmilos** [comment]
M & S are not raws strictly speaking. There was already a request to track these, don't know why the user closed it: https://github.com/darktable-org/rawspeed/issues/442
